File: leetcode/two-sum.go
func twoSum(nums []int, target int) []int {
	mem := make(map[int]int)

	for i := 0; i < len(nums); i++ {
		complement := target - nums[i]
		val, ok := mem[complement]
		if ok {
			return []int{val, i}
		}
		mem[nums[i]] = i
	}

	return nil
}

File: leetcode/two-sum.scala
object Solution {
    def twoSum(nums: Array[Int], target: Int): Array[Int] = {
        var mem = scala.collection.mutable.Map[Int, Int]()
        
        for(i <- 0 until nums.length) {
            val remainder = target - nums(i)
            if(mem.contains(remainder)) {
                return Array(mem(remainder), i)
            } else {
                mem(nums(i)) = i
            }
        }
        
        Array(-1, -1)
    }
}
